### ISSUE TITLE
Training with a dataset that does not provide mmaction

### ISSUE THREAD

**Issue by antonio2600** [open-mmlab/mmaction2#2097]
Hello.
To to test a new model with a dataset made by me, I have to do exactly what is shown in the example but changing the paths of data_root, data_root_val, ann_file_train, etc?
Can I use the cfg used in the example or do I have to create one myself?
I have the same question for the file load_from = './checkpoints/tsn_r50_1x1x3_100e_kinetics400_rgb_20200614-e508be42.pth'

![Captura desde 2022-11-27 17-01-52](https://user-images.githubusercontent.com/42941507/204145051-721c902f-cc9a-4d37-bac3-268f2c9dbf38.png)

**Comment by hukkai** [comment]
@antonio2600 

If you only need to test a single video, you can refer to the demo: https://github.com/open-mmlab/mmaction2/tree/master/demo 

If you only need to test a model on new datasets, you just need to modify configs beginning with `cfg.data.test` and configs for train and val will not be used. You need to prepare the dataset in the same formate as the dataset the config is using. For example, Kinetics data folder structure can be found here: https://github.com/open-mmlab/mmaction2/tree/master/tools/data/kinetics

**Comment by antonio2600** [comment]
I want to train a new model from 0 and test it later. I already have the dataset prepared, the videos divided into train, test and validation. I have been trying to follow the example of https://github.com/open-mmlab/mmaction2/blob/master/demo/mmaction2_tutorial.ipynb and it gives me an error. I have changed all the cfg and nothing.
I have the annotation files too: train.txt, test.txt, validation.txt and the label map.

**Comment by antonio2600** [comment]
![image](https://user-images.githubusercontent.com/42941507/204370295-24d545f6-2fe6-4de2-b086-30de39c09390.png)

**Comment by antonio2600** [comment]
And this is my code:
![Captura desde 2022-11-28 21-04-36](https://user-images.githubusercontent.com/42941507/204370491-7b582bd2-0f8b-429c-99eb-5901a959a438.png)

**Comment by antonio2600** [comment]
And if I use the code that provides the example of using a new dataset, it works perfetly.
![image](https://user-images.githubusercontent.com/42941507/204370862-bdd38211-c6c8-4088-b4e1-20364b342e60.png)

**Comment by hukkai** [comment]
This is because you set the number of classes as 2 but your dataset has more than 2 classes @antonio2600

**Comment by antonio2600** [comment]
Ok. I have changed the nombur of classes, I have 31 classes. Now the error is that:

![Captura desde 2022-11-29 12-00-49](https://user-images.githubusercontent.com/42941507/204512531-bf9fd8e9-103b-4c9b-99b7-fa9c9dd91e0c.png)

**Comment by hukkai** [comment]
The error is still that your labels are the same or larger than the number of classes. Please note that the label should begin with 0.

**Comment by antonio2600** [comment]
Ok, I have began with 1 instead 0. I'm going to trying it now.

**Comment by antonio2600** [comment]
Another question. If I have the dataset divided into train, test and validation. How do I have to modify the code to use the 3 datasets?

**Comment by hukkai** [comment]
Use `ann_file_train`, `ann_file_val` and `ann_file_test` to store the data list of these dataset.

**Comment by antonio2600** [comment]
And once I have trained the model with my dataset. How can I test with some videos this trained model?
This is the result of the training, maybe I have to do some adjustments with the dataset or try some technique to improve the result.
![Captura desde 2022-11-30 12-20-20](https://user-images.githubusercontent.com/42941507/204783215-d26b85e0-5bc6-4be1-8f40-4b0d63df87f2.png)

**Comment by hukkai** [comment]
You can refer to the document https://mmaction2.readthedocs.io/en/latest/getting_started.html?highlight=test#test-a-dataset

**Comment by antonio2600** [comment]
Yes but the question that I have is once a model is trained with a new dataset, like the code that I show below, where this model is stored?, the checkpoints file, tsn_r50_1x1x3_100e_kinetics400_rgb_20200614-e508be42.pth, is it updated?

![Captura desde 2022-11-30 19-30-43](https://user-images.githubusercontent.com/42941507/204879557-6e275dcb-398e-4bb3-88fc-5d2e6950b4f8.png)

**Comment by hukkai** [comment]
The saved checkpoint is at `./work_dirs` by default.

**Comment by antonio2600** [comment]
Is the checkpoint latest.pth?

![Captura desde 2022-12-01 00-33-35](https://user-images.githubusercontent.com/42941507/204930654-0e0d4fe4-eabc-43f7-bc22-54426b9abdb4.png)

**Comment by hukkai** [comment]
yes

**Comment by antonio2600** [comment]
Ok. Thank you so much for everything.

**Comment by antonio2600** [comment]
If I want to test a single video, what do I have to select in config_file?
Because my config is diferent that kinetics400_rgb.py, do I have to create a new file?

![Captura desde 2022-12-01 13-50-37](https://user-images.githubusercontent.com/42941507/205057395-c04ce35f-6124-43c1-b863-db634cf18d69.png)

**Comment by antonio2600** [comment]
I have tried to use a config and initialize the recognizer like the example, but something is wrong.

![Captura desde 2022-12-01 14-09-34](https://user-images.githubusercontent.com/42941507/205061321-917f4726-cc1a-4e1e-869a-2df08d49407f.png)


![Captura desde 2022-12-01 14-10-05](https://user-images.githubusercontent.com/42941507/205061264-21d6df4c-a39f-42cb-a413-750086b8bdc1.png)

**Comment by antonio2600** [comment]
The label_map.txt is this:

![Captura desde 2022-12-01 19-55-00](https://user-images.githubusercontent.com/42941507/205136523-8e5155f7-7307-4251-8456-d81efba984d2.png)

**Comment by z-w-wang** [comment]
Do you have the pre-trained resnet50 model checkpoint file on kinetics and pre-trained bn-inception model checkpoint file? 
The README.md checkpoint page is 404 now.
If have, could you share it to me?

**Comment by antonio2600** [comment]
I have the tsn_r50_1x1x3_100e_kinetics400 checkpoint file only, do you want it?

**Comment by z-w-wang** [comment]
> I have the tsn_r50_1x1x3_100e_kinetics400 checkpoint file only, do you want it?

Sure! Thanks a loooooooooooooooooooooooot!

**Comment by antonio2600** [comment]
Yes but I try to send to you the file and github doesn't support that file type because it's .pth. How do we do it?

**Comment by z-w-wang** [comment]
My homepage has my email. Maybe we can try this! 
Or upload it on Google Drive or BaiduYun? Do whatever is convenient for you. ^_^

**Comment by antonio2600** [comment]
I just sent it to you by drive.

**Comment by z-w-wang** [comment]
Google Drive says that I need access permissions, and I submitted an application to you.

**Comment by antonio2600** [comment]
Hello, I forgot to give you permissions, I have already given them to you. Check it.

**Comment by z-w-wang** [comment]
Got it! Thx once again! 
What is the powerpoint in email? Forgive me for being so ignorant that I cannot understand this language, I guess it may be a summary powerpoint about a paper?

**Comment by antonio2600** [comment]
Your welcome!
I don't know what are you telling about the powerpoint in email. I only sent you the file. Have you recived another document?
